src/backend/distributed/master/master_delete_protocol.c

```diff
@@ -498,6 +568,157 @@ ShardsMatchingDeleteCriteria(Oid relationId, List *shardIntervalList,
 
 
 /*
+ * SendQueryToShards executes the given query in all placements of the given 
+ * shard list and returns the total affected tuple count. The execution is done
+ * in a distributed transaction and the commit protocol is decided according to
+ * the value of citus.multi_shard_commit_protocol parameter. SendQueryToShards
+ * does not acquire locks for the shards so it is advised to acquire locks to
+ * the shards when necessary.
+ */
+static int
+SendQueryToShards(Query *query, List *shardIntervalList)
+{
+	ListCell *shardIntervalCell = NULL;
+	List *connectionList = NIL;
+	int affectedTupleCount = 0;
+
+	PG_TRY();
+	{
+		foreach(shardIntervalCell, shardIntervalList)
+		{
+			ShardInterval *shardInterval = (ShardInterval *) lfirst(
+				shardIntervalCell);
+			Oid relationId = shardInterval->relationId;
+			uint64 shardId = shardInterval->shardId;
+			StringInfo shardQueryString = makeStringInfo();
+			char* shardQueryStringData = NULL;
+			
+			deparse_shard_query(query, relationId, shardId, shardQueryString);
+			shardQueryStringData = shardQueryString->data;
+			int shardAffectedTupleCount = SendQueryToPlacements(shardInterval, 
+																shardQueryStringData,
+																&connectionList);
+			affectedTupleCount += shardAffectedTupleCount;
+		}
+
+		if (MultiShardCommitProtocol == COMMIT_PROTOCOL_2PC)
+		{
+			PrepareRemoteTransactions(connectionList);
+		}
+	}
+	PG_CATCH();
+	{
+		/* roll back all transactions */
+		AbortRemoteTransactions(connectionList);
+		CloseConnections(connectionList);
+
+		PG_RE_THROW();
+	}
+	PG_END_TRY();
+
+	CommitRemoteTransactions(connectionList, false);
+
+	return affectedTupleCount;
+}
+
+
+/*
+ * SendQueryToPlacement sends a query to all placement of a shard. Before the actual
+ * query is sent, 
+ */
+static int
+SendQueryToPlacements(ShardInterval *shardInterval, char *shardQueryString, 
+					  List **connectionList)
+{
+	uint64 shardId = shardInterval->shardId;
+	List *shardPlacementList = FinalizedShardPlacementList(shardId);;
+	List *shardFailedPlacementList = NIL;
+	int32 shardAffectedTupleCount = -1; // TODO-Eren: -1 or 0
+	ListCell *shardPlacementCell = NULL;
+	
+	foreach(shardPlacementCell, shardPlacementList)
+	{
+		ShardPlacement *shardPlacement = (ShardPlacement *) lfirst(
+			shardPlacementCell);
+		char *workerName = shardPlacement->nodeName;
+		uint32 workerPort = shardPlacement->nodePort;
+		char *nodeUser = CurrentUserName();
+		PGconn *connection = ConnectToNode(workerName, workerPort, nodeUser);
+		TransactionConnection *transactionConnection = NULL;
+		PGresult *result = NULL;
+		char *placementAffectedTupleString = NULL;
+		int32 placementAffectedTupleCount = -1;
+
+		if (connection == NULL)
+		{
+			shardFailedPlacementList = lappend(shardFailedPlacementList,
```

Comment:
Since the logic to update the shard placement state is no longer there, we can just remove this and error out if something fails.